configuration/uart.md

```diff
@@ -89,6 +89,28 @@ By default, the primary UART is assigned to the Linux console. If you wish to us
 1. At the prompt `Would you like the serial port hardware to be enabled?` answer 'Yes'
 1. Exit raspi-config and reboot the Pi for changes to take effect.
 
+## Enabling early console (earlycon) for Linux
+
+Although the Linux kernel starts the UARTs relatively early in the boot process, it is still long after some critical bits of infrastructure have been set up. A failure in those early stages can be hard to diagnose without access to the kernel log messages from that time. That's the problem that the "earlycon" mechanism was created to work around. Consoles that support earlycon usage present an additional interface to the kernel that allows for simple, synchronous output - printk won't return until the characters have been output to the UART.
```

Comment:
"printk won't return until the characters have been output to the UART." - does that mean that using earlycon slows down the kernel boot speed? (or is the difference negligible?)

Reply:
If you need earlycon then you really don't give a damn how long it takes. It will slow booting very slightly, but it very soon switches to a regular console so I don't think it's worth mentioning.